src/remote.c

```diff
@@ -670,21 +670,47 @@ int git_remote_set_pushurl(git_repository *repo, const char *remote, const char*
 	return set_url(repo, remote, CONFIG_PUSHURL_FMT, url);
 }
 
-const char* git_remote__urlfordirection(git_remote *remote, int direction)
+static int resolve_url(git_buf *resolved_url, const char *url, int direction, const git_remote_callbacks *callbacks)
 {
-	assert(remote);
+	int status;
+
+	if (callbacks && callbacks->resolve_url) {
+		git_buf_clear(resolved_url);
+		status = callbacks->resolve_url(resolved_url, url, direction, callbacks->payload);
+		if (status != GIT_PASSTHROUGH) {
+			git_error_set_after_callback_function(status, "git_resolve_url_cb");
+			git_buf_sanitize(resolved_url);
+			return status;
+		} else {
+			/* reset passthrough */
+			status = 0;
```

Comment:
This is never used

Reply:
Fixed!

Reply:
Original author was lost :(

Reply:
Restored, sorry about that (GitX hard-rewrites amends, and I keep not fixing that bug).

Reply:
Just finishing up on branch graphs, identity management is next in my client (beta.gitfox.app). If you are on macOS, maybe check it out ;)